src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs

```diff
@@ -193,6 +201,111 @@ private List<string> ResolveByExternalReferencePackages(List<string> uidList, Co
+                    }
+                }
+                Logger.LogInfo($"{uidList.Count - unresolvedUidList.Count} uids found in {requestUrl}.");
+                uidList = new List<string>(unresolvedUidList);
+                unresolvedUidList.Clear();
+            }
+
+            Logger.LogInfo($"{needQueryCount - uidList.Count} uids found in xrefservice.");
+            return uidList;
+        }
+
+        private async Task<Tuple<bool, List<XRefSpec>>> QueryByHttpRequestAsync(string requestUrl, List<string> smallPiece)
+        {
+            using (var client = new HttpClient())
+            {
+                var failedResult = new Tuple<bool, List<XRefSpec>>(false, null);
+                try
+                {
+                    client.BaseAddress = new Uri(requestUrl);
+                }
+                catch (UriFormatException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return failedResult;
+                }
+                catch (ArgumentException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return failedResult;
+                }
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                StringContent content = new StringContent(JsonUtility.Serialize(smallPiece), System.Text.Encoding.UTF8, "application/json");
+                HttpResponseMessage response = null;
+                try
+                {
+                    response = await client.PostAsync("", content);
+                }
+                catch (HttpRequestException e)
+                {
+                    Logger.LogWarning(e.InnerException.Message + "\n" + $"Failed to resolve {smallPiece.Count} uids from {requestUrl}: for example including " + smallPiece.Take(10).ToDelimitedString());
+                    return failedResult;
+                }
+
+                if (response.StatusCode == System.Net.HttpStatusCode.OK)
+                {
+                    var data = await response.Content.ReadAsStreamAsync();
+                    List<XRefSpec> xsList;
+                    using (var sr = new StreamReader(data))
+                    {
+                        try
+                        {
+                            xsList = JsonUtility.Deserialize<List<XRefSpec>>(sr);
+                        }
+                        catch (Newtonsoft.Json.JsonReaderException e)
+                        {
+                            Logger.LogWarning($"Response from {requestUrl} is not in valid JSON format." + e.Message);
+                            return failedResult;
+                        }
+                    }
+                    return new Tuple<bool, List<XRefSpec>>(true, xsList);
```

Comment:
any difference?

---
In reply to: [127985337](https://github.com/dotnet/docfx/pull/1837#discussion_r127985337) [](ancestors = 127985337)